sky/serve/service_spec.py

```diff
@@ -39,14 +43,26 @@ def __init__(
             with ux_utils.print_exception_no_traceback():
                 raise ValueError('readiness_path must start with a slash (/). '
                                  f'Got: {readiness_path}')
+
+        if qps_upper_threshold is not None or qps_lower_threshold is not None:
+            with ux_utils.print_exception_no_traceback():
+                raise ValueError(
+                    'qps_upper_threshold and qps_lower_threshold are '
+                    'deprecated. Please use target_qps_per_replica instead.')
+
+        if auto_restart is not None:
+            with ux_utils.print_exception_no_traceback():
+                raise ValueError('auto_restart is deprecated.')
+
```

Comment:
Could we somehow combine these three checks?

Reply:
I think these are two separate checks.